### ISSUE TITLE
Check `dataset` is an allowed value in `load_variables`?

### ISSUE THREAD

**Issue by mikemahoney218** [walkerke/tidycensus#419]
I make this mistake (using an incorrect data set code) a lot:

```r
> vars <- tidycensus::load_variables(2019, "acs")
Error: lexical error: invalid char in json text.
                                       <html><head><title>Error report
                     (right here) ------^
```

This is entirely my fault! The `dataset` argument is well documented with the list of allowed options. But this error message always confuses me for a second until I look up the help file again.

Would you be interested in checking to make sure `dataset` is one of the allowed options before sending the request? Something along the lines of:

```r
allowed_datasets <- c("sf1", "sf3", "acs1", "acs3", "acs5", "acs1/profile", "acs3/profile", "acs5/profile", "acs1/subject", "acs3/subject", "acs5/subject")
if (length(dataset) != 1 && 
    !(dataset %in% allowed_datasets)) {
    stop("`dataset` must be one (and only one) of: ",
         paste0(allowed_datasets, collapse = ", "))
} 
```

Also, an incredibly minor typo I just noticed -- `acs3/profile` is missing a `"` here:
https://github.com/walkerke/tidycensus/blob/b9100d88bcb20a62ad022bd66848682f7dfa3041/R/search_variables.R#L8

Thanks for the awesome package!

**Comment by walkerke** [comment]
Yeah that's a good idea and we actually support more datasets than that now, including the 2020 PL, SF2 and SF4 in 2010, and the island area datasets in 2010.  I'll get this updated.

**Comment by walkerke** [comment]
Implemented in #456, thanks to @lpiep!
